documentation/docs/pkcs11/luks.md

```diff
@@ -0,0 +1,407 @@
+default-hierarchy=unified
+```
+
+Unfortunately, Ubuntu 22.04 does not provide p11-kit support, however the setup works fine for
+Ubuntu 23.10.
+
+#### 1. Install the `p11-kit` package.
+
+*Ubuntu 23.10*
+
+```bash
+sudo apt install p11-kit
+```
+
+*RHEL 9*
+
+```bash
+sudo dnf install p11-kit
+```
+
+#### 2. Create the PKCS#11 configuration and module directories
+
+```bash
+sudo mkdir -p /etc/pkcs11/modules
+```
+
+#### 3. Create a configuration file for the PKCS#11 module.
+
+```bash
+sudo tee /etc/pkcs11/pkcs11.conf <<EOF
+# This setting controls whether to load user configuration from the
+# ~/.config/pkcs11 directory. Possible values:
+#    none: No user configuration
+#    merge: Merge the user config over the system configuration (default)
+#    only: Only user configuration, ignore system configuration
+user-config: merge
+EOF
+```
+
+#### 4. Copy the PKCS#11 module to the pkcs11 directory.
+
+```bash
+sudo cp libckms_pkcs11.so /usr/local/lib/
+```
+
+#### 5. Create a configuration file for the ckms PKCS#11 module.
+
+```bash
+sudo tee /etc/pkcs11/modules/ckms_pkcs11.module <<EOF
+# Cosmian KMS PKCS#11 module
+module: /usr/local/lib/libckms_pkcs11.so
+EOF
+```
+
+#### 6. Check that the module loads correctly.
+
+```bash
+> p11-kit list-modules
+
+...
+ckms_pkcs11: /usr/local/lib/libckms_pkcs11.so
+ library-description: Cosmian KMS PKCS#11 provider
+ library-manufacturer: Cosmian
+ library-version: x.y
+ token: Cosmian-KMS
+     manufacturer: Cosmian
+     model: software
+     serial-number: x.y.z
+     flags: 
+           rng
+           write-protected
+           login-required
+           user-pin-initialized
+           protected-authentication-path
+           token-initialized
+
+```
+
+## Configuring the access to the KMS
+
+The PKCS#11 module uses the same configuration file as the [CLI](../cli/cli.md#configuration).
+Since it may be run as a system user, the configuration file should be made available
+in `/etc/cosmian/kms.json`.
+
+See [Authenticating users to the KMS](../authentication.md) to learn
+how to configure the KMS to use Open ID connect or certificate authentication.
+
+Here is an example configuration file for the PKCS#11 provider library accessing the KMS using a
+PKCS#12 file for authentication.
+
+```json
+{
+  "kms_server_url": "https://kms.acme.com:9999",
+  "ssl_client_pkcs12_path": "./certificates/machine123.acme.p12",
+  "ssl_client_pkcs12_password": "machine123_pkcs12_password"
+}
+```
+
+To use Open ID connect, install the `ckms` CLI from
+[Cosmian packages](https://package.cosmian.com/kms/) and
+use the `ckms login` command to authenticate to the KMS first.
+
+## Creating an RSA key pair using openssl and importing it into the Cosmian KMS
+
+To generate a self-signed certificate with RSA 2048bit key and in PKCS12 format, you can use the
+OpenSSL command-line tool. Here are the steps:
+
+#### 1. Generate a new private key:
```

Comment:
wrong section levels?

Reply:
On purpose. The docs template does not make sufficient visual difference between the second and third level